### PR TITLE
IME text composition box deactivated

### PR DESCRIPTION
**Pull request by wstagg**
The function that is used to make the IME text composition box has been deactivated. 

The function and corrosponding code has been left in place for a potential future fix or re-implementation.

### CODE REVIEW

**Comment by mrmbernardi** [comment]
I think if you're going to remove it, you should remove the entire thing. The point of version control is that with a bit of searching we can go back and find the original code even after its been removed from the repo.

You should also probably combine this with the work to fix up the location of the system drawn composition box.

@IntelOrca, do you think this function is still needed?

**Comment by wstagg** [comment]
Yes I agree with fully removing the function.

I shall do that and investigate the position of the system drawn box as well.

**Comment by wstagg** [comment]
I have moved 
`SDL_Rect rect = { cursorX, cursorY, 100, 100 };
        SDL_SetTextInputRect(&rect);
        SDL_StartTextInput();`

from TextComposition.cpp to TextInput.cpp. The positon of the system drawn IME box now follows the position of the text box the player is typing in,

**Comment by Gymnasiast** [comment]
I think it would be good to have a user who uses a CJK language to test this, as it is going to affect them. @telk5093 perhaps something you’d like to test?

**Comment by telk5093** [comment]
I need a pre-built one to test, I don't have enough time to build his branch myself. Please let me know what should I check it.

**Comment by AaronVanGeffen** [comment]
> I need a pre-built one to test, I don't have enough time to build his branch myself. Please let me know what should I check it.

You can download a CI build from the actions tab :)

**Comment by wstagg** [comment]
To test you need to open up a text box (I have been using the save game and new file option or changing the name of the park etc).

**Comment by telk5093** [comment]
I've tested it. It is impossible to write any text (while Ctrl+C,V works).

1. It is able to write text when just open the game, but the changes are not applied
2. After close text boxes, both Korean and roman characters are not written.

**Comment by wstagg** [comment]
https://github.com/OpenRCT2/OpenRCT2/assets/103323985/5a042af6-d67a-4622-8228-42c51a62673a

I dont think I am having the same issue. Could this be an OS difference?

**Comment by wstagg** [comment]
OKay yeah I see that it doesnt like changing back and forth between languages. I think I see where its going wrong.

**Comment by wstagg** [comment]
OKay yeah I see that it doesnt like changing back and forth between languages. As you can see in the clip I can swap from english to japanese but it doesnt like the swap back to english from Japanese. I have a rough idea of where this could be going wrong so I shall investigate. 

@telk5093 what OS are you running?

[IME.webm](https://github.com/OpenRCT2/OpenRCT2/assets/103323985/14ca7403-ae1b-47c2-9665-0e84bd4b5e10)

**Comment by telk5093** [comment]
> @telk5093 what OS

Windows 10.

Note that it was impossible to change both Eng=>Kor and Kor=>Eng for me.

**Comment by wstagg** [comment]
Okay noted. I shall get back to you after some investigation. Thanks

**Comment by tupaschoal** [comment]
Hi @wstagg, did you get a chance to come back to this?

**Comment by wstagg** [comment]
Hello @tupaschoal, very sorry, life got very busy. Let me get back to it!

**Comment by tupaschoal** [comment]
No problem!

**Comment by wstagg** [comment]
RIght this seems to be working for me now. There was a bug that @telk5093 mentioned where you could no longer write any text after changing from english to korean/japanese and then back to english. I beleive this to be fixed now.

**Comment by tupaschoal** [comment]
Hi @telk5093 , did you get a chance to try it?

**Comment by telk5093** [comment]
> Hi @telk5093 , did you get a chance to try it?

Sorry I have some private issues. Might take times

**Comment by tupaschoal** [comment]
No problem, take your time

**Comment by telk5093** [comment]
It seems that works fine for Hangeul

**Review by tupaschoal — APPROVED** [review]

**Review by Gymnasiast — COMMENTED** [review]
I’m going off @telk5093 ’s testing, as I am not familiar with IMEs.

This PR will need a changelog entry.

**Comment by wstagg** [comment]
> I’m going off @telk5093 ’s testing, as I am not familiar with IMEs.
> 
> 
> 
> This PR will need a changelog entry.

@Gymnasiast is this something I need to do?

**Comment by karst** [comment]
> Is this something I need to do?

Yes please

**Comment by tupaschoal** [comment]
@wstagg you need to edit this file with a description of what it fixes and the issue.
https://github.com/OpenRCT2/OpenRCT2/blob/develop/distribution%2Fchangelog.txt

**Comment by ZehMatt** [comment]
You need to rebase your branch, you are a bit behind.

**Comment by Gymnasiast** [comment]
The entry is also out of order - it needs to be sorted by type (Fix) first, then by ID.

We can help you rebase if necessary.

File: distribution/changelog.txt [hunk 1]
```diff
@@ -23,6 +23,7 @@
 - Fix: [#21198] [Plugin] Setting brake or booster speeds on a tile element doesn’t work.
 - Fix: [#21290] Sound keeps playing when paused from fast-forward mode.
 - Fix: [#21291] Hungry guests heading to any flat ride do not count for warning threshold (original bug).
+- Fix: [#20631] IME window not positioned correctly.
```
Comment (Gymnasiast) [inline distribution/changelog.txt:26]: As I mentioned already, the changelog should be sorted by type first, then by ID. 20631 is lower than 21291.

Comment (wstagg) [reply distribution/changelog.txt]: Sorry about that... Got there eventually!

**Review by tupaschoal — APPROVED** [review]
Seems ok to me. @Gymnasiast ?

**Review by Gymnasiast — APPROVED** [review]

**Comment by mrmbernardi** [comment]
I'm seeing two issues with this fix. I am using Fcitx5 and Mozc.

Firstly the IME window appears to be offset quite a distance below the caret. I think this might be caused by the height of the given `SDL_rect`.

Secondly, the position doesn't take into account UI scaling. 

See the following images:

ime1
ime2

**Comment by wstagg** [comment]
Where would you say the IME box should be positioned?

I didn't think it should cover the text?

image

**Comment by mrmbernardi** [comment]
The usual position is directly under the text that currently being edited. Also I just noticed that usually the currently selected option is put into the textbox as well, but perhaps this isn't always necessary

ime3

**Comment by wstagg** [comment]
Ah yes. Perhaps I should make some more changes then to adjust the position of the box and look into the scaling?

**Pull request merged**